docs/proposals/multi-node/README.md

```diff
@@ -0,0 +1,623 @@
+      ...
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector: {} # podSelector that matches the member pods of the given etcd cluster instance
+            topologyKey: "topology.kubernetes.io/zone
+      ...
+    ...
+  ...
+
+```
+
+The recommendation is to keep `etcd-druid` agnostic of such topics related scheduling and cluster-topology and to use [kupid](https://github.com/gardener/kupid) to [orthogonally inject](https://github.com/gardener/kupid#mutating-higher-order-controllers) the desired [pod anti-affinity](https://github.com/gardener/kupid/blob/master/config/samples/cpsp-pod-affinity-anti-affinity.yaml).
+
+#### Alternative
+
+Another option is to build the functionality into `etcd-druid` to include the required pod anti-affinity when it provisions the `StatefulSet` that manages the member pods.
+While this has the advantage of avoid a dependency on an external component like [kupid](https://github.com/gardener/kupid), the disadvantage is that such built-in support necessarily limits what kind of topologies of the underlying cluster will be supported.
+Hence, it is better to keep `etcd-druid` altogether agnostic of issues related to scheduling and cluster-topology.
+
+### PodDisruptionBudget
+
+This proposal recommends that `etcd-druid` should deploy [`PodDisruptionBudget`](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets) (`maxUnavailable` set to `floor(<cluster size>/2)`) for multi-node etcd clusters to ensure that any planned disruptive operations can try and honour the disruption budget to ensure high availability of the etcd cluster.
+
+## Rolling updates to etcd members
+
+Any changes to the `Etcd` resource spec that might result in a change to `StatefulSet` spec or otherwise result in a rolling update of member pods should be applied/propagated by `etcd-druid` only when the etcd cluster is fully healthy to reduce the risk of quorum loss during the updates.
+If the cluster is unhealthy, `etcd-druid` must restor it to full health before proceeding with the rolling update.
+This can be further optimized in the future to handle the cases where rolling updates can still be performed on an etcd cluster that is not fully healthy.
+
+## Follow Up
+
+### Shoot Control-Plane Migration
+
+This proposal adds support for multi-node etcd clusters but it should not have significant impact on [shoot control-plane migration](https://github.com/gardener/gardener/blob/master/docs/proposals/07-shoot-control-plane-migration.md) any more than what already present in the single-node etcd cluster scenario.
+But to be sure, this needs to be discussed further.
+
+### Performance impact of multi-node etcd clusters
+
+Multi-node etcd clusters incur a cost on [write performance](https://etcd.io/docs/v2/admin_guide/#optimal-cluster-size) as compared to single-node etcd clusters.
+This performance impact needs to be measured and documented.
```

Comment:
...and balanced vs. the possibly lower latency and higher throughput of local or RAM disks?